Openapiplugin can not handle nullable enums

### Expected behavior

As a CobiGen user, I want the openapiplugin to be able to handle nullable enums.

### Actual behavior
When using a yaml file with a nullable enum, CobiGen throws a NPE.

```
Unable to invoke public abstract java.util.List<com.devonfw.cobigen.impl.config.entity.Trigger> com.devonfw.cobigen.impl.generator.api.TriggerMatchingEvaluator.getMatchingTriggers(java.lang.Object).
Unable to invoke public abstract java.util.List<com.devonfw.cobigen.impl.config.entity.Trigger> com.devonfw.cobigen.impl.generator.api.TriggerMatchingEvaluator.getMatchingTriggers(java.lang.Object).
com.devonfw.cobigen.api.exception.CobiGenRuntimeException: Unable to invoke public abstract java.util.List<com.devonfw.cobigen.impl.config.entity.Trigger> com.devonfw.cobigen.impl.generator.api.TriggerMatchingEvaluator.getMatchingTriggers(java.lang.Object).
at com.devonfw.cobigen.impl.util.ExceptionUtil.invokeTarget(ExceptionUtil.java:91)
at com.devonfw.cobigen.impl.aop.CachedInterceptor.invoke(CachedInterceptor.java:57)
at com.sun.proxy.$Proxy48.getMatchingTriggers(Unknown Source)
at com.devonfw.cobigen.impl.generator.ConfigurationInterpreterImpl.getMatchingTemplatesConfigurations(ConfigurationInterpreterImpl.java:172)
at com.devonfw.cobigen.impl.generator.ConfigurationInterpreterImpl.getMatchingTemplates(ConfigurationInterpreterImpl.java:87)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.devonfw.cobigen.impl.util.ExceptionUtil.invokeTarget(ExceptionUtil.java:82)
at com.devonfw.cobigen.impl.aop.CachedInterceptor.invoke(CachedInterceptor.java:57)
at com.sun.proxy.$Proxy47.getMatchingTemplates(Unknown Source)
at com.devonfw.cobigen.impl.generator.CobiGenImpl.getMatchingTemplates(CobiGenImpl.java:168)
at com.devonfw.cobigen.eclipse.generator.AnalyzeInputJob.run(AnalyzeInputJob.java:70)
at org.eclipse.jface.operation.ModalContext$ModalContextThread.run(ModalContext.java:122)
Caused by: java.lang.reflect.InvocationTargetException
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.base/java.lang.reflect.Method.invoke(Method.java:566)
at com.devonfw.cobigen.impl.util.ExceptionUtil.invokeTarget(ExceptionUtil.java:82)
... 14 more
Caused by: java.lang.NullPointerException
at com.devonfw.cobigen.openapiplugin.inputreader.OpenAPIInputReader.extractProperties(OpenAPIInputReader.java:404)
at com.devonfw.cobigen.openapiplugin.inputreader.OpenAPIInputReader.extractComponents(OpenAPIInputReader.java:234)
at com.devonfw.cobigen.openapiplugin.inputreader.OpenAPIInputReader.getInputObjects(OpenAPIInputReader.java:131)
at com.devonfw.cobigen.impl.generator.TriggerMatchingEvaluatorImpl.getMatchingTriggers(TriggerMatchingEvaluatorImpl.java:86)
... 19 more
```

### Steps to reproduce (bug)

1. Create a *.yaml file with following input:

```
openapi: 3.0.0
servers:
  - url: 'https://localhost:8081/server/services/rest'
info:
  title: Devon Example
  description: Example of a API definition
  version: 1.0.0
paths:
  /tablemanagement/v1/table/{id}:
    get:
      operationId: findTable
      parameters:
        - name: id
          in: path
          required: true
          schema:
            type: integer
            format: int64
            minimum: 0
            maximum: 50
      responses:
        '200':
          description: Any
components:
    schemas:
        Table:
          x-component: tablemanagament
          description: Entity definiton of Table
          type: object
          properties:
            myenum: 
               type: string
               nullable: true
               enum: 
                  - enum1
                  - null
```
3. Start the "Generate" process. 


### Affected version:
openapiplugin 7.0.0

--- a/cobigen/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/main/java/com/devonfw/cobigen/openapiplugin/inputreader/OpenAPIInputReader.java
+++ b/cobigen/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/main/java/com/devonfw/cobigen/openapiplugin/inputreader/OpenAPIInputReader.java
@@ -401,7 +401,11 @@ public class OpenAPIInputReader implements InputReader {
 
                 List<String> enumElements = new ArrayList<>();
                 for (Object element : enums) {
-                    enumElements.add(element.toString());
+                    if (element != null) {
+                        enumElements.add(element.toString());
+                    } else {
+                        enumElements.add("null");
+                    }
                 }
                 propModel.setEnumElements(enumElements);
             }

--- a/cobigen/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/test/java/com/devonfw/cobigen/openapiplugin/unittest/inputreader/OpenAPIInputReaderTest.java
+++ b/cobigen/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/test/java/com/devonfw/cobigen/openapiplugin/unittest/inputreader/OpenAPIInputReaderTest.java
@@ -279,6 +279,29 @@ public class OpenAPIInputReaderTest {
         List<Object> inputObjects = getInputs("invalidPath.yaml");
     }
 
+    /**
+     * Tests if the input reader can handle nullable enums. See:
+     * https://github.com/devonfw/cobigen/issues/1244
+     *
+     * @throws Exception
+     */
+    @Test
+    public void testNullableEnum() throws Exception {
+        List<Object> inputObjects = getInputs("nullableEnum.yaml");
+
+        for (Object o : inputObjects) {
+            if (isEntityDef(o)) {
+                EntityDef entity = (EntityDef) o;
+                List<PropertyDef> properties = entity.getProperties();
+                for (PropertyDef p : properties) {
+                    List<String> enums = p.getEnumElements();
+                    assertThat(enums.get(0)).isEqualTo("enum1");
+                    assertThat(enums.get(1)).isEqualTo("null");
+                }
+            }
+        }
+    }
+
     @Test(expected = InvalidConfigurationException.class)
     public void testInvalidXComponent() throws Exception {
         List<Object> inputObjects = getInputs("invalidXComponent.yaml");
@@ -412,6 +435,16 @@ public class OpenAPIInputReaderTest {
      */
     private boolean isComponentDef(Object o) {
         return o.getClass() == ComponentDef.class;
+    }
+
+    /**
+     * Checks whether the object is an {@link EntityDef}
+     * @param o
+     *            object to check whether it is an EntityDef
+     * @return true if it is an EntityDef
+     */
+    private boolean isPropertyDef(Object o) {
+        return o.getClass() == PropertyDef.class;
     }
 
 }

--- a/cobigen/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/test/resources/testdata/unittest/OpenAPIInputReaderTest/nullableEnum.yaml
+++ b/cobigen/cobigen-openapiplugin-parent/cobigen-openapiplugin/src/test/resources/testdata/unittest/OpenAPIInputReaderTest/nullableEnum.yaml
@@ -1,0 +1,36 @@
+openapi: 3.0.0
+servers:
+  - url: 'https://localhost:8081/server/services/rest'
+info:
+  title: Devon Example
+  description: Example of a API definition
+  version: 1.0.0
+paths:
+  /tablemanagement/v1/table/{id}:
+    get:
+      operationId: findTable
+      parameters:
+        - name: id
+          in: path
+          required: true
+          schema:
+            type: integer
+            format: int64
+            minimum: 0
+            maximum: 50
+      responses:
+        '200':
+          description: Any
+components:
+    schemas:
+        Table:
+          x-component: tablemanagament
+          description: Entity definiton of Table
+          type: object
+          properties:
+            myenum: 
+               type: string
+               nullable: true
+               enum: 
+                  - enum1
+                  - null

